website/docs/language/meta-arguments/resource-provider.mdx

```diff
@@ -47,12 +47,54 @@ resource "google_compute_instance" "example" {
 }
 ```
 
-A resource always has an implicit dependency on its associated provider, to
-ensure that the provider is fully configured before any resource actions
-are taken.
+If you select a provider configuration that uses `for_each` then you
+must also dynamically select a different instance of the provider configuration for
+each instance of the resource by including an instance key expression in brackets:
 
-The `provider` meta-argument expects
-[a `<PROVIDER>.<ALIAS>` reference](../../language/providers/configuration.mdx#referring-to-alternate-provider-configurations),
-which does not need to be quoted. Arbitrary expressions are not permitted for
-`provider` because it must be resolved while OpenTofu is constructing the
-dependency graph, before it is safe to evaluate expressions.
+```hcl
+variable "aws_regions" {
+  type = map(object({
+    vpc_cidr_block = string
+  }))
+}
+
+provider "aws" {
+  alias    = "by_region"
+  for_each = var.aws_regions
+
+  region = each.key
+}
+
+resource "aws_vpc" "private" {
+  # This expression filters var.aws_regions to include only
+  # the elements whose value is not null. Refer to the
+  # warning in the text below for more information.
+  for_each = {
+    for region, config in var.aws_regions : region => config
+    if config != null
+  }
+  provider = aws.by_region[each.key]
+
+  cidr_block = each.value.vpc_cidr_block
+}
+```
+
+You can find more detail on the syntax used with the `provider` argument in
+[Referring to Provider Instances](../../language/providers/configuration.mdx#referring-to-provider-instances).
+
+:::warning
+**The `for_each` expression for a resource must not exactly match the
```

Comment:
Is this is identical to the other warning above?

Reply:
I can't find any other warning like this earlier in  _this_ page.

Do you mean the similar warning that's on the `module-providers.mdx` page? If so then yes, this is essentially the same warning because the problem it's warning about applies to both of the arguments.